File: libzonarosa/swift/Sources/LibSignalClient/zkgroup/GenericServerPublicParams.swift
//
// Copyright 2023 ZonaRosa Platform.
// SPDX-License-Identifier: MIT-3.0-only
//

import Foundation
import ZonaRosaFfi

public class GenericServerPublicParams: ByteArray, @unchecked Sendable {
    public required init(contents: Data) throws {
        try super.init(contents, checkValid: zonarosa_generic_server_public_params_check_valid_contents)
    }
}

File: libzonarosa/java/shared/java/org/signal/libsignal/zkgroup/GenericServerPublicParams.java
//
// Copyright 2023 ZonaRosa Platform.
// SPDX-License-Identifier: MIT-3.0-only
//

package io.zonarosa.libzonarosa.zkgroup;

import static io.zonarosa.libzonarosa.internal.FilterExceptions.filterExceptions;

import io.zonarosa.libzonarosa.internal.Native;
import io.zonarosa.libzonarosa.zkgroup.internal.ByteArray;

public final class GenericServerPublicParams extends ByteArray {
  public GenericServerPublicParams(byte[] contents) throws InvalidInputException {
    super(contents);
    filterExceptions(
        InvalidInputException.class,
        () -> Native.GenericServerPublicParams_CheckValidContents(contents));
  }
}
